MEN-4324: integration test for the file transfer feature

tranchitella opened the pull request:
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

tranchitella commented:
@mender-test-bot start pipeline

tranchitella commented:
@mender-test-bot start pipeline

tranchitella commented:
Failures in the "mender-connect bogus protocol" are expected, everything else is green, including the file transfer tests.

lluiscampos commented on tests/tests/test_filetransfer.py:
```diff
@@ -0,0 +1,190 @@
+            assert r.headers.get("X-Men-File-Uid") == "0"
+            assert r.headers.get("X-Men-File-Mode") == "600"
+            assert r.headers.get("X-Men-File-Path") == "/tmp/random.bin"
+            assert r.headers.get("X-Men-File-Size") != ""
```
Can't we check the exact size in this case?

lluiscampos commented on tests/tests/test_filetransfer.py:
```diff
@@ -0,0 +1,190 @@
+import hashlib
+import io
+import os
+import os.path
```
We can remove this import, as we already have `os`

lluiscampos reviewed (approved):
Looks very good. The test logic is well expressed, and I haven't spotted any obvious error.

I recall @merlin-northern being concern about parallel downloads of files; do you guys think we can test this scenario here or are we confident enough about the protocol now?

merlin-northern commented:
> Looks very good. The test logic is well expressed, and I haven't spotted any obvious error.
> 
> I recall @merlin-northern being concern about parallel downloads of files; do you guys think we can test this scenario here or are we confident enough about the protocol now?

this is something I want to test some more manually, and probably a ticket is coming, so I would for now keep it as is, if it proves to be a bug, then with the ticket resolution we will extend the test, if it makes sense.

kacf commented on tests/tests/test_filetransfer.py:
```diff
@@ -0,0 +1,190 @@
+                authtoken,
+                mode="600",
+                uid="0",
+                gid="0",
```
Suggestion: Use some random mode/uid/gid values here, which will test that both the upload and the download of these work.

kacf reviewed (approved):
Very good test!

tranchitella commented:
@kacf @lluiscampos I addressed your comments here: https://github.com/mendersoftware/integration/pull/1256/commits/9e7d6cfc946903f56c6d72c0d06311e808cec498
@mender-test-bot start pipeline, please

tranchitella commented:
@mender-test-bot start pipeline

tranchitella commented:
@mender-test-bot start pipeline

tranchitella commented:
Manually launched pipeline (because of issues in GitLab, it seems):
https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/270604329

tranchitella commented:
Failures in `test_demo_artifact` and `test_bogus_proto_message` are spurious.
